azurerm/data_source_batch_pool.go

```diff
@@ -0,0 +1,159 @@
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"publisher": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"offer": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"sku": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"version": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+					},
+				},
+				Set: resourceArmVirtualMachineStorageImageReferenceHash,
+			},
+			"node_agent_sku_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmBatchPoolRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).batchPoolClient
+
+	name := d.Get("name").(string)
+	accountName := d.Get("account_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	ctx := meta.(*ArmClient).StopContext
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: Batch pool %q in account %q (Resource Group %q) was not found", name, accountName, resourceGroup)
+		}
+		return fmt.Errorf("Error making Read request on AzureRM Batch pool %q: %+v", name, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", name)
+	d.Set("account_name", accountName)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("vm_size", resp.VMSize)
+
+	if resp.ScaleSettings != nil {
+		if resp.ScaleSettings.AutoScale != nil {
+			d.Set("scale_mode", azure.BatchPoolAutoScale)
+			d.Set("autoscale_evaluation_interval", resp.ScaleSettings.AutoScale.EvaluationInterval)
+			d.Set("autoscale_formula", resp.ScaleSettings.AutoScale.Formula)
+		} else if resp.ScaleSettings.FixedScale != nil {
+			d.Set("scale_mode", azure.BatchPoolFixedScale)
+			d.Set("target_dedicated_nodes", resp.ScaleSettings.FixedScale.TargetDedicatedNodes)
+			d.Set("target_low_priority_nodes", resp.ScaleSettings.FixedScale.TargetLowPriorityNodes)
+			d.Set("resize_timeout", resp.ScaleSettings.FixedScale.ResizeTimeout)
+		}
+	}
+
+	if resp.DeploymentConfiguration != nil &&
+		resp.DeploymentConfiguration.VirtualMachineConfiguration != nil &&
+		resp.DeploymentConfiguration.VirtualMachineConfiguration.ImageReference != nil {
+
+		imageReference := resp.DeploymentConfiguration.VirtualMachineConfiguration.ImageReference
+
+		if err := d.Set("storage_image_reference", schema.NewSet(resourceArmVirtualMachineStorageImageReferenceHash, flattenAzureRmBatchPoolImageReference(imageReference))); err != nil {
+			return fmt.Errorf("[DEBUG] Error setting AzureRM Batch Pool Image Reference: %#v", err)
```

Comment:
I think we can remove the debug here
```suggestion
			return fmt.Errorf("Error setting AzureRM Batch Pool Image Reference: %#v", err)
```

Reply:
Ok.